docs: Implement style guide rules - active voice, misc (#23272)
MystenLabs/sui: docs/content/concepts/cryptography/system/checkpoint-verification.mdx

---
title: Checkpoint Verification
description: On the Sui network, checkpoints define the history of the blockchain. Checkpoint verification is how full nodes and other clients guarantee their state matches the Sui network.
---

On the Sui network, checkpoints define the blockchain history. Unlike traditional blockchains that create blocks before execution, Sui creates checkpoints after transaction execution to provide a certified record of chain history.

Checkpoints contain:

- The cryptographic hash of the previous checkpoint  
- A list of all transaction digests and their corresponding transaction effects digests included in the checkpoint  
- A set of signatures from a quorum (more than two-thirds) of the validators that formed the committee when the checkpoint was created  

Validators and full nodes consume checkpoints to remain synchronized with the network.

## Verifying checkpoints

> For full nodes and validators to trust a checkpoint, they must verify it. Verification ensures that the checkpoint is authentic and created by the Sui validator committee.

Reviewer: technically passive

> For full nodes and validators to trust a checkpoint, they must verify it. Verification ensures that the checkpoint is authentic and created by the Sui validator committee.

Author: ```suggestion
Full nodes and validators verify a checkpoint to trust it. This verification confirms that the Sui validator committee created it and that it is authentic.
```
